pkg/controller/validate/validate_test.go

```diff
@@ -190,28 +190,22 @@ func TestClusterValidation(t *testing.T) {
 	})
 	t.Run("Valid cluster specs", func(t *testing.T) {
 		t.Run("Advanced cluster spec specified", func(t *testing.T) {
-			spec := mdbv1.AtlasDeploymentSpec{AdvancedDeploymentSpec: &mdbv1.AdvancedDeploymentSpec{}, DeploymentSpec: nil}
-			assert.NoError(t, DeploymentSpec(&spec, false, "NONE"))
-			assert.Nil(t, DeploymentSpec(&spec, false, "NONE"))
-		})
-		t.Run("Regular cluster specs specified", func(t *testing.T) {
-			spec := mdbv1.AtlasDeploymentSpec{AdvancedDeploymentSpec: nil, DeploymentSpec: &mdbv1.DeploymentSpec{}}
-			assert.NoError(t, DeploymentSpec(&spec, false, "NONE"))
-			assert.Nil(t, DeploymentSpec(&spec, false, "NONE"))
-		})
-
-		t.Run("Serverless Cluster", func(t *testing.T) {
-			spec := mdbv1.AtlasDeploymentSpec{AdvancedDeploymentSpec: nil, DeploymentSpec: &mdbv1.DeploymentSpec{
-				ProviderSettings: &mdbv1.ProviderSettingsSpec{
-					ProviderName: "SERVERLESS",
-				},
-			}}
+			spec := mdbv1.AtlasDeploymentSpec{DeploymentSpec: &mdbv1.AdvancedDeploymentSpec{}, ServerlessSpec: nil}
 			assert.NoError(t, DeploymentSpec(&spec, false, "NONE"))
 			assert.Nil(t, DeploymentSpec(&spec, false, "NONE"))
 		})
+		// t.Run("Serverless Cluster", func(t *testing.T) {
```

Comment:
This test shows that it is possible to set a serverless deployment using a legacy deployment, I think it's safe to remove